skimage/io/_io.py

```diff
@@ -53,8 +53,16 @@ def imread(fname, as_grey=False, plugin=None, flatten=None,
     with file_or_url_context(fname) as fname:
         img = call_plugin('imread', fname, plugin=plugin, **plugin_args)
 
-    if as_grey and getattr(img, 'ndim', 0) >= 3:
-        img = rgb2grey(img)
+    if not hasattr(img, 'ndim'):
+        return img
+
+    if img.ndim > 2:
+        if img.shape[-1] not in (3, 4) and img.shape[-3] in (3, 4):
```

Comment:
why ``img.shape[-3]`` and not ``img.shape[0]``?

Reply:
It needs to access the metadata. Otherwise some labeling annotations might get it wrong when doing the segmentation for object detection applications. The labels where rotated 90º when having vertical images. It took me a while to find out where was the bug :(   ... That's what the metadata is for. We shouldn't assume anything about the shape. We need certainty.

I would suggest to use exiftool. 

Regards

Reply:
@JavierClearImageAI wow, this issue is a blast from the past! And yet, as you noted, very much still an issue.

The main problem with metadata is that it is inconsistent between different file formats, or even different variants of the same format. We don't have enough person-hours to pursue all the possible formats and metadata tags, and the lack of clarity over where to start has prevented real efforts from getting started here. See the following issues in imageio for more context:

https://github.com/imageio/imageio/issues?q=is%3Aissue+is%3Aopen+metadata

imageio is the right place to start thinking about and contributing to a solution.

Thank you for the ping!